test/TestHelpers.jl

```diff
@@ -198,6 +198,24 @@ end
     val
 end
 
+@inline function Base.deleteat!(A::OffsetArray, i::Int)
+    checkbounds(A, i)
```

Comment:
Actually, my suggestion aimed at requiring less changes, not more. If that involves doing changes elsewhere then I agree it's not the place to do it.

Anyway, just add the `@boundscheck` annotations and it should be good to merge, though a quick review of the `OffsetArray` parts by @timholy would be nice.